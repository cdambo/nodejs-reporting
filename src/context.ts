export default class Context {
  private readonly globalTags: object;

  public constructor(globalTags: object = {}) {
    this.globalTags = globalTags;
  }

  public mergeTags(tags: object = {}): object {
    return Object.assign({}, this.globalTags, tags);
  }
}
