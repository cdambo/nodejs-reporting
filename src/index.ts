import Context from "./context";

export * from "./metrics";
export { Context };
