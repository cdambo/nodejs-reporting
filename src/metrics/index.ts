export * from "./formats";
export * from "./reporters";
export * from "./integrations";
