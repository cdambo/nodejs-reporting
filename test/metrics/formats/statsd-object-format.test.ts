import { StatsDObjectFormat } from "../../../src";

describe("StatsDObjectFormat", (): void => {
  describe("format", (): void => {
    it("Formats the metric to an object", (): void => {
      expect(
        StatsDObjectFormat().format({
          metric: "increment",
          stat: "metric_name",
          tags: { a: 1, b: 2 }
        })
      ).toMatchSnapshot();
    });

    it("Formats the metric with options", (): void => {
      expect(
        StatsDObjectFormat({
          transformations: {
            metric: (m: string): string => `${m}_FORMATTED`,
            stat: (s: string): string => `${s}_FORMATTED`,
            tags: (t: object): object => ({ ...t, thisIs: "FORMATTED" }),
            args: (a: object): object => ({ ...a, thisIs: "FORMATTED" })
          }
        }).format({
          metric: "increment",
          stat: "metric_name",
          tags: { a: 1, b: 2 },
          arg1: "123",
          arg2: 456
        })
      ).toMatchSnapshot();
    });

    it("Formats the tags to snake case keys", (): void => {
      expect(
        StatsDObjectFormat().format({
          metric: "increment",
          stat: "metric_name",
          tags: {
            camelCaseTag1: "tagOne",
            camelCaseTag2: "tag_two",
            anotherTag: 5
          }
        })
      ).toMatchSnapshot();
    });

    it("Removes leading non-letters from metric name", (): void => {
      expect(
        StatsDObjectFormat().format({
          metric: "increment",
          stat: "123._-/*()#$@!a_real_metric"
        })
      ).toMatchSnapshot();
    });

    it("Replaces with an underscore characters that are not ASCII alphanumerics, underscores, or periods in metric name", (): void => {
      expect(
        StatsDObjectFormat().format({
          metric: "increment",
          stat: "a-real_Metric.Name_with+bad#chars!@-#$%^&*()9"
        })
      ).toMatchSnapshot();
    });
  });
});
