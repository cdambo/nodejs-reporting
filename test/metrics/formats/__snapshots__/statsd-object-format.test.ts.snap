// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsDObjectFormat format Formats the metric to an object 1`] = `
Object {
  "increment": Object {
    "stat": "metric_name",
    "tags": Object {
      "a": 1,
      "b": 2,
    },
  },
}
`;

exports[`StatsDObjectFormat format Formats the metric with options 1`] = `
Object {
  "increment_FORMATTED": Object {
    "arg1": "123",
    "arg2": 456,
    "stat": "metric_name_FORMATTED",
    "tags": Object {
      "a": 1,
      "b": 2,
      "thisIs": "FORMATTED",
    },
    "thisIs": "FORMATTED",
  },
}
`;

exports[`StatsDObjectFormat format Formats the tags to snake case keys 1`] = `
Object {
  "increment": Object {
    "stat": "metric_name",
    "tags": Object {
      "another_tag": 5,
      "camel_case_tag_1": "tagOne",
      "camel_case_tag_2": "tag_two",
    },
  },
}
`;

exports[`StatsDObjectFormat format Removes leading non-letters from metric name 1`] = `
Object {
  "increment": Object {
    "stat": "a_real_metric",
    "tags": Object {},
  },
}
`;

exports[`StatsDObjectFormat format Replaces with an underscore characters that are not ASCII alphanumerics, underscores, or periods in metric name 1`] = `
Object {
  "increment": Object {
    "stat": "a_real_Metric.Name_with_bad_chars___________9",
    "tags": Object {},
  },
}
`;
